src/mam4xx/aero_model_emissions.hpp

```diff
@@ -0,0 +1,972 @@
+  }
+
+  // set dust_dmt_vwr ....
+  for (int i = 0; i < dust_nbin; ++i) {
+    dust_dmt_vwr[i] = 0.0;
+  }
+
+  // Bin physical properties
+  // [frc] Geometric std dev PaG77 p. 2080 Table1
+  Real gsd_anl = 2.0;
+  Real ln_gsd = haero::log(gsd_anl);
+
+  // Set a fundamental statistic for each bin
+  // Mass median diameter analytic She84 p.75 Table 1 [m]
+  // FIXME: FIXME: this appears to be a bug--vma is set, and then re-set
+  //        maybe the second one should be dmt_nma?
+  Real dmt_vma = 2.524e-6;
+  dmt_vma = 3.5e-6;
+  // Compute analytic size statistics
+  // Convert mass median diameter to number median particle diameter [m]
+  Real dmt_nma = dmt_vma * haero::exp(-3.0 * haero::square(ln_gsd));
+
+  // [m] Size Bin minima
+  Real sz_min[sz_nbr];
+  // [m] Size Bin maxima
+  Real sz_max[sz_nbr];
+  // [m] Size Bin centers
+  Real sz_ctr[sz_nbr];
+  // [m] Size Bin widths
+  Real sz_dlt[sz_nbr];
+
+  // [m3 m-3] Volume concentration resolved
+  Real vlm_rsl[dust_nbin] = {0.0};
+
+  // Compute resolved size statistics for each size distribution
+  // In C. Zender's code call dst_sz_rsl
+  for (int n = 0; n < dust_nbin; ++n) {
+    // Factor for logarithmic grid
+    Real series_ratio = haero::pow((dmt_max[n] / dmt_min[n]), (1.0 / sz_nbr));
+    sz_min[0] = dmt_min[n];
+    // NOTE: Loop starts at 1 (2 in fortran code)
+    for (int m = 1; m < sz_nbr; ++m) {
+      sz_min[m] = sz_min[m - 1] * series_ratio;
+    }
+    // Derived grid values
+    // NOTE: Loop ends at sz_nbr - 2 (fortran: sz_nbr - 1)
+    for (int m = 0; m < sz_nbr - 1; ++m) {
+      sz_max[m] = sz_min[m + 1];
+    }
+    // set the final entry
+    sz_max[sz_nbr - 1] = dmt_max[n];
+    // Final derived grid values
+    for (int m = 0; m < sz_nbr; ++m) {
+      sz_ctr[m] = 0.5 * (sz_min[m] + sz_max[m]);
+      sz_dlt[m] = sz_max[m] - sz_min[m];
+    }
+
+    constexpr Real pi = haero::Constants::pi;
+    // Lognormal distribution at sz_ctr
+    // Factor in lognormal distribution
+    // NOTE: original variable name in fortran: lngsdsqrttwopi_rcp
+    Real lnN_factor = 1.0 / (ln_gsd * haero::sqrt(2.0 * pi));
+
+    for (int m = 0; m < sz_nbr; ++m) {
+      // Evaluate lognormal distribution for these sizes (call lgn_evl)
+      Real tmp = haero::log(sz_ctr[m] / dmt_nma) / ln_gsd;
+      Real lgn_dst =
+          lnN_factor * haero::exp(-0.5 * haero::square(tmp)) / sz_ctr[m];
+      Real coeff = pi / 6.0 * haero::pow(sz_ctr[m], 3) * lgn_dst * sz_dlt[m];
+      // Integrate moments of size distribution
+      dust_dmt_vwr[n] += sz_ctr[m] * coeff;
+      vlm_rsl[n] += coeff;
+    } // end for (m)
+    // Mass weighted diameter resolved [m]
+    dust_dmt_vwr[n] /= vlm_rsl[n];
+  } // end for (dust_nbin)
+} // end init_dust_dmt_vwr()
+
+KOKKOS_INLINE_FUNCTION
+void dust_emis(
+    // in
+    const int dust_indices[dust_nbin + dust_nnum], const Real dust_density,
+    const Real (&dust_flux_in)[dust_nflux_in], const DustEmissionsData data,
```

Comment:
@mjs271 , Can you explain to me what this input does: `const Real (&dust_flux_in)[dust_nflux_in]`? What is different from the input `const Real dust_flux_in[dust_nflux_in]`?.

Reply:
As I understand it to work, `Real (&var)[var_size]` passes a reference to the array of `Real`s. This is in contrast to the version without parentheses, `Real &var[var_size]` that passes an array of references to `Real`s and is [categorically illegal in C++](https://stackoverflow.com/questions/1164266/why-are-arrays-of-references-illegal).

From a practical standpoint, I was trying to do the illegal one and ran across this method for making it work. Whether passing by reference is even worth it here (or a good idea)... well that's another question 🙃

All that being said, I'm happy to change it if it's dumb or causes confusion